ndcube/global_coords/global_coords.py

```diff
@@ -0,0 +1,76 @@
+from collections.abc import Mapping
+
+from ndcube import NDCube
+
+__all__ = ['GlobalCoords']
+
+
+class GlobalCoords(Mapping):
+    """
+    A structured representation of coordinate information applicable to a whole NDCube.
+    """
+    def __init__(self, NDCube):
+        super().__init__()
+
+        # Set values using the setters for validation
+        self.mapping = {}
+
+    def add(self, name, physical_type):
+        """
+        Add a new coordinate to the collection.
+        """
+        if len(self.mapping) > 1:
```

Comment:
I'm not sure this will work.  What if the `GlobalCoords` instance is empty?  Also it looks like you haven't created a link between `physical_type` and `name`.  For instance, if I want to know the `physical_type` of `name`, there doesn't appear to be a way to do that.  Finally, I don't see where the coord value is stored.  Looks like that's been left out?  I'll add some more discussion in the PR conversation.

Reply:
Yeah, you are right... I think I was just thinking about special cases only. Let me amend this shortly.

Reply:
Removed the unnecessary constraint

Reply:
Or condition...